src/plans/stepsize.jl

```diff
@@ -1687,3 +1687,193 @@ end
 function get_last_stepsize(step::WolfePowellBinaryLinesearchStepsize, ::Any...)
     return step.last_stepsize
 end
+
+@doc raw"""
+    DistanceOverGradientsStepsize{R<:Real} <: Stepsize
+
+Implements the Riemannian Distance over Gradients (RDoG) stepsize schedule, a learning-rate-free 
+optimization method for Riemannian manifolds introduced by [@dodd2024learning].
+
+This stepsize adapts automatically without hyperparameter tuning by tracking the maximum distance 
+from the initial point and accumulating gradient norms.
+
+# Fields
+
+* `initial_distance::R`: initial distance estimate ``ϵ > 0``
+* `max_distance::R`: maximum distance from initial point ``\bar{r}_t = \max_{s≤t} d(p_0, p_s)``
+* `gradient_sum::R`: accumulated squared gradient norms ``G_t = \sum_{s=0}^t \|∇f(p_s)\|^2``
+* `initial_point`: the initial point ``p_0``
+* `use_curvature::Bool`: whether to use sectional curvature in the stepsize
+* `sectional_curvature_bound::R`: lower bound on sectional curvature ``κ`` (only used if `use_curvature=true`)
+
+The stepsize at iteration ``t`` is computed as:
+
+```math
+η_t = \frac{\bar{r}_t}{\sqrt{ζ_κ(\bar{r}_t)} \sqrt{G_t}}
+```
+
+where ``ζ_κ`` is the geometric curvature function:
+- ``ζ_κ(d) = 1`` if ``κ ≥ 0`` or `use_curvature=false`
+- ``ζ_κ(d) = \frac{\sqrt{|κ|} d}{\tanh(\sqrt{|κ|} d)}`` if ``κ < 0``
+
+# Constructor
+
+    DistanceOverGradientsStepsize(M::AbstractManifold; kwargs...)
+
+Initialize the RDoG stepsize on manifold `M`.
+
+## Keyword arguments
+
+* `initial_distance=1e-3`: initial distance estimate ``ϵ``
+* `use_curvature=false`: whether to incorporate sectional curvature
+* `sectional_curvature_bound=0.0`: lower bound on sectional curvature (if known)
+$(_var(:Keyword, :p; add="initial point, used to track distances"))
+
+# See also
+
+[`AdaptiveWNGradientStepsize`](@ref), [`gradient_descent`](@ref)
+"""
+mutable struct DistanceOverGradientsStepsize{R <: Real, P} <: Stepsize
+    initial_distance::R
+    max_distance::R
+    gradient_sum::R
+    initial_point::P
+    use_curvature::Bool
+    sectional_curvature_bound::R
+    last_stepsize::R
+end
+
+function DistanceOverGradientsStepsize(
+        M::AbstractManifold;
+        p = rand(M),
+        initial_distance::R = 1.0e-3,
+        use_curvature::Bool = false,
+        sectional_curvature_bound::R = 0.0,
+    ) where {R <: Real}
+    return DistanceOverGradientsStepsize{R, typeof(p)}(
+        initial_distance,
+        initial_distance,  # max_distance starts at initial_distance
+        zero(R),          # gradient_sum starts at 0
+        copy(M, p),       # store initial point
+        use_curvature,
+        sectional_curvature_bound,
+        NaN,              # last_stepsize
+    )
+end
+
+"""
+    geometric_curvature_function(κ::Real, d::Real)
+
+Compute the geometric curvature function ``ζ_κ(d)`` for sectional curvature bound ``κ``.
+
+```math
```

Comment:
One could either do a `raw"""` string to avoid the double `\\`,
or use interpolation and the tricks I started in the `documentation_glossary`.

The first is easier, the second complicated but a bit more flexible in the end.

If you feel this is too much to do, I can also refactor this later.